### ISSUE TITLE
JS: Hoisting in Javascript

### ISSUE THREAD

**Issue by arp99** [girlscript/winter-of-contributing#161]
# Hoisting in JavaScript
Discuss the concept of hoisting in JavaScript.

## Directory
```
Topics/6.Hoisting
```
## Description

- Talk about What is Hoisting
- How javascript uses hoisting for declarations
- Effect Of Hoisting on different types of Declarations

**Comment by arp99** [comment]
Hello Supervisor/ Mentor, Can I start working on this issue I just created? 😅

**Comment by paramsiddharth** [comment]
> Hello Supervisor/ Mentor, Can I start working on this issue I just created? 😅

@arp99 Yes, please! Go ahead. I might require you to rename the folder's serial number, but it is a small change and we'll deal
with that later.

**Comment by arp99** [comment]
Yeah, The serial is the only thing that is needed to be changed in the future, Thanks !  😅

**Comment by paramsiddharth** [comment]
> Yeah, The serial is the only thing that is needed to be changed in the future, Thanks ! 😅

No worries! It is a small thing, I'll do it on my own if the need be! :) Best of luck!

**Comment by Ananyagupta43** [comment]
@paramsiddharth Can I start working on this topic ?

**Comment by 7sakshi7** [comment]
I am a GWOC'21 contributor from Batch 3 of JavaScript. Please assign this issue to me. I assure you that this issue's title will be the same as my PR's title as well as I will be Contributing under the Guidelines and Code of Conduct.
Discord : agarwalSakshi | Contributor#9567

**Comment by anjalirai12** [comment]
Sir, I would like to work on this issue , So kindly, please assign it to me.

**Comment by paramsiddharth** [comment]
@7sakshi7 @anjalirai12 Please note that we already had a document contributed for this, so you may either extend it or contribute a different format. Best of luck!

**Comment by anjalirai12** [comment]
@paramsiddharth Sir , I have extended the JS hoisting topic ,added some more important information about the topic. So please kindly review it.

**Comment by paramsiddharth** [comment]
@Moeen-Ul-Islam Would you like to be assigned?

**Comment by Moeen-Ul-Islam** [comment]
@paramsiddharth  I don't mind!
